### ISSUE TITLE
Mat/Pat - Emails 

### ISSUE THREAD

**Issue by fazil-ey** [bcgov/digital-journeys#874]
The following emails need to be sent as the applicant goes through the mat/pat journey 

---------------------------------------------------------------------------------
##### 1.1 Email to applicant work email confirming submission. There should be no pdf attachment

> **Sub: Maternity/Parental Leave Application Received**
> This is to confirm that we have received your application for maternity/parental leave. Your application has been forwarded to your supervisor for review and approval. Once your supervisor approves your leave, we will notify you via email. You can also check the status of your application in the "Submitted Forms" tab on [Digital Journeys](https://digitaljourneys.apps.silver.devops.gov.bc.ca/form)
>

##### 1.2 Email to supervisor informing about a new submission for approval. There should be no pdf attachment (both eligible & non eligible)

> **Sub: [Employee Name]'s Maternity/Parental Leave Application Received For Approval** 
>This is to inform you that a new maternity/parental leave application has been submitted by [Employee Name] and is ready for your review and approval. The application can be found in the "Review & Process" tab on [Digital Journeys](https://digitaljourneys.apps.silver.devops.gov.bc.ca/form)
>
>The requested start date for the leave is [start date]. We recommend that you approve this application as soon as possible, but no later than 4 weeks before the start date. This will allow timely processing of the application and prevent payroll errors.

#####   1.3 Email to applicant informing about a resubmission after editing. There should be no pdf attachment. Also note that no email is to be sent to the supervisor in case of resubmission if applicant edits before supervisor approval

> **Sub: Maternity/Parental Leave Application Received**
This is to confirm that we have received your application for maternity/parental leave. 
Your application has been forwarded to your supervisor for review and approval. Once your supervisor approves your leave, we will notify you via email. You can also check the status of your application in the "Submitted Forms" tab on [Digital Journeys](https://digitaljourneys.apps.silver.devops.gov.bc.ca/form)

##### 2.1 Email to applicant work email confirming approval. Attach pdf in email
_NOTE_ in Case 1, 2, and 3 in the Next Step section there will be a instruction about updating benefits. This instruction should change based on Employee Class. The copy & logic for the benefits is described below.

> **Benefits instructions to be added to Next Steps based on Employee Class**
> 
> **If employee class is BCGEU or PEA then** 
> 
> - Add you baby to your benefits within 60 days of the birth of your baby. To do so -
>     - Fill out the following forms & send them to you supervisor
>        - [Bargaining Unit Benefits Program Enrolment/Change Form](https://www2.gov.bc.ca/assets/gov/careers/all-employees/pay-and-benefits/benefits-for-bargaining-unit-employees/bargaining_unit_benefits_enrolment_change.pdf) 
>        - [Group Life Beneficiary Designation](https://www2.gov.bc.ca/assets/gov/careers/forms-tools/all-employees/group_life_beneficiary_designation.pdf) (if applicable) 
>        - [Evidence of Insurability](https://www2.gov.bc.ca/assets/gov/careers/forms-tools/all-employees/evidence_of_insurability_bu.pdf) (if applicable)
>     - Ask your supervisor to submit an AskMyHR service request for "My Team or Organization" under the following category "Benefits > Submit a Health Benefit Form/Application"
> - You can learn more about benefit on [Careers & MyHR](https://www2.gov.bc.ca/gov/content/careers-myhr/all-employees/pay-benefits/benefits/bargaining-benefits)
> 
> **If employee class is Excluded**  
> - Add you baby to your benefits within 60 days of the birth of your baby. To do so -
>     - Fill out the following forms & send them to you supervisor
>        - [Flexible Benefits Enrolment/Change Form](https://www2.gov.bc.ca/assets/gov/careers/all-employees/pay-and-benefits/benefits-for-excluded-employees/flexible_benefits_enrolment_change.pdf)
>        - [Group Life Beneficiary Designation](https://www2.gov.bc.ca/assets/gov/careers/forms-tools/all-employees/group_life_beneficiary_designation.pdf) (if applicable) 
>        - [Evidence of Insurability](https://www2.gov.bc.ca/assets/gov/careers/forms-tools/all-employees/evidence_of_insurability_bu.pdf) (if applicable)
>     - Ask your supervisor to submit an AskMyHR service request for "My Team or Organization" under the following category "Benefits > Submit a Health Benefit Form/Application"
> - You can learn more about benefits on [Careers & MyHR](https://www2.gov.bc.ca/gov/content/careers-myhr/all-employees/pay-benefits/benefits/excluded)
> 

 **2.1.1 Case 1: The applicant _did not_ apply for allowance but is eligible**
>  **Sub: Maternity/Parental Leave Application Approved**
This is to confirm that your application for maternity/parental leave has been approved. You can find a pdf of your application attached in the email. Please save the copy for future reference. You can also find your application in the "Submitted Forms" tab on [Digital Journeys](https://digitaljourneys.apps.silver.devops.gov.bc.ca/form)
>
> **Next Steps** 
> - [_Benefits instruction based on Employee Class_]
> - After you go on leave, your supervisor will be your point of contact. Please direct any
questions or concerns regarding your leave to your supervisor. 

   **2.1.2 Case 2: The applicant applied for allowance**
> **Sub: Maternity/Parental Leave Application Approved**
This is to confirm that your application for maternity/parental leave has been approved. You can find a pdf of your application attached in the email. Please save the copy for future reference. You can also find your application in the "Submitted Forms" tab on [Digital Journeys](https://digitaljourneys.apps.silver.devops.gov.bc.ca/form)
>
> **Next Steps** 
> - To receive your allowance, **you must apply for [Employment Insurance](https://www.canada.ca/en/services/benefits/ei/ei-maternity-parental/apply.html)** after you start your leave. After your EI is approved, you must provide proof of approval and send it to the supervisor. To do so you need to - 
>     - Log into the Service Canada website. Click on “My Latest Claim".
>     - Take a picture or screenshot of the contents on ["My Latest Claim" page](https://images.zenhubusercontent.com/62e986a3ff9c07df27cce462/6420bdc4-e678-4348-a9c9-94f63d22a233). This information confirms that the you have been approved by Service Canada to receive the Maternity/Parental benefits. 
>     - Send this to your supervisor and ask them to create a [AskMyHR](https://sfs7.gov.bc.ca/affwebservices/public/saml2sso?SPID=urn:ca:bc:gov:customerportal:prod) service request for "My Team or Organization" under the following category "Leave & Time Off > Maternity, Parental, Adoption".
>      - [_only if taking extended parental leave allowance_] Since you opted for extended parental leave allowance, you must provide the proof of approval again after you start your parental leave. You should complete the above steps again after you start your parental leave to receive your parental leave allowance. 
> - [_Benefits instruction based on Employee Class_]
> - [_only if taking parental leave_] Once your baby's birth certificate is available, send a copy of the birth certificate to your supervisor. This is needed to confirm your eligibility to take parental leave. Your supervisor must create an [AskMyHR](https://sfs7.gov.bc.ca/affwebservices/public/saml2sso?SPID=urn:ca:bc:gov:customerportal:prod) service request on your behalf to provide the birth certificate to the payroll team. The service request must be created under the service category - "Leave & Time Off > Maternity, Parental, Adoption" 
   > - After you go on leave, your supervisor will be your point of contact. Please direct any questions or concerns regarding your leave to your supervisor. 

  **2.1.3 Case 3: The applicant deferred allowance** 
  > **Sub: Maternity/Parental Leave Application Approved**
This is to confirm that your application for maternity/parental leave has been approved. You can find a pdf of your application attached in the email. Please save the copy for future reference. You can also find your application in the "Submitted Forms" tab on [Digital Journeys](https://digitaljourneys.apps.silver.devops.gov.bc.ca/form)

  > **Next Steps** 
  > - To claim your allowance after you return to work, **you must apply for [Employment Insurance](https://www.canada.ca/en/services/benefits/ei/ei-maternity-parental/apply.html)** after you start your leave. After your EI is approved, please save a copy of your proof of approval. You will need it to claim your allowance after you return to work. To save a copy - 
>     - Log into the Service Canada website. Click on “My Latest Claim".
>     - Take a picture or screenshot of the contents on ["My Latest Claim" page](https://images.zenhubusercontent.com/62e986a3ff9c07df27cce462/6420bdc4-e678-4348-a9c9-94f63d22a233). This information confirms that the you have been approved by Service Canada to receive the Maternity/Parental benefits. 
> - [_Benefits instruction based on Employee Class_]
> - [_only if taking parental leave_] Once your baby's birth certificate is available, send a copy of the birth certificate to your supervisor. This is needed to confirm your eligibility to take parental leave. Your supervisor must create an [AskMyHR](https://sfs7.gov.bc.ca/affwebservices/public/saml2sso?SPID=urn:ca:bc:gov:customerportal:prod) service request on your behalf to provide the birth certificate to the payroll team. The service request must be created under the service category - "Leave & Time Off > Maternity, Parental, Adoption" 
> - After you go on leave, your supervisor will be your point of contact. Please direct any questions or concerns regarding your leave to your supervisor. 

 **2.1.4 Case 4: The applicant is not eligible for benefits**
> **Sub: Maternity/Parental Leave Application Approved**
This is to confirm that your application for maternity/parental leave has been approved. You can find a pdf of your application attached in the email. Please save the copy for future reference. You can also find your application in the "Submitted Forms" tab on [Digital Journeys](https://digitaljourneys.apps.silver.devops.gov.bc.ca/form).
After you go on leave, your supervisor will be your point of contact. Please direct any questions or concerns regarding your leave to your supervisor. 

#####  2.2 (OR) Email to applicant informing application sent back
  > **Sub: Maternity/Parental Leave Application Sent Back**
This is to inform you that your application for maternity/parental leave has been sent back by your supervisor. Please reach out to your supervisor to learn more. You can edit and resubmit the application again for approval from the "Submitted Forms" tab on [Digital Journeys](https://digitaljourneys.apps.silver.devops.gov.bc.ca/form)

---------------------------------------------------------------------------------
#####  3.1 Email to applicant informing about supervisor's change. Attach pdf in email
  > **Sub: Maternity/Parental Leave Application Edited By Supervisor**
This is to inform you that your application for maternity/parental leave has been edited by your supervisor. If this is not expected, please reach out to your supervisor to learn more. You can find a pdf of your application attached in the email. Please save a copy for future reference.

#####  3.1 Email to supervisor confirming change. Attach pdf in email
   > **Sub: [Employee Name] 's Maternity/Parental Leave Application Edited Successfully**
This is to inform you that [Employee Name]'s application for maternity/parental leave has been edited successfully. You can find a pdf of the application attached in the email. Please save the copy for future reference. The application can be found in the "Review & Process" tab on [Digital Journeys](https://digitaljourneys.apps.silver.devops.gov.bc.ca/form)

#### 4. Reminder to return to work. [#1218](https://app.zenhub.com/workspaces/digital-journeys-61c10fe84c8bcb001491c5f5/issues/gh/bcgov/digital-journeys/1218)
Sent to applicant's alternate email. To be sent 6 weeks before end of leave (last date of maternity and/or parental leave).  

> **Sub: Reminder to notify return to work 4 weeks before Maternity/Parental Leave **
This is to inform you that you must inform your supervisor about your intent to return to work at least 4 weeks before your intended date of return. Please include the following information i your communication to your supervisor 
    - Type of leave you took
    - First day of work 
    - Phone number for supervisor to connect with you if needed
You must ask your supervisor to submit an AskMyHR service request for "My Team or Organization" under the following category "Leave & Time Off > Maternity, Parental, Adoption" with the letter attached.

>
>You can also inform your supervisor about your intent to extend your leave if you do not intend to return to work on your originally chosen return date. To extend your leave, your should ask your supervisor to update your Maternity and Parental leave form with the new length of maternity or parental leave and resubmit the form.
>
> _[only if deferred allowance]_ (Copy To be confirmed)
>  To claim your allowance after you return to work, you should have applied for [Employment Insurance](https://www.canada.ca/en/services/benefits/ei/ei-maternity-parental/apply.html) after you started your leave. You will need a copy of your proof of approval. To get your proof of approval - 
>  
>    -  Log into the Service Canada website. Click on “My Latest Claim".
>    - Take a picture or screenshot of the contents on ["My Latest Claim" page](https://images.zenhubusercontent.com/62e986a3ff9c07df27cce462/6420bdc4-e678-4348-a9c9-94f63d22a233). This information confirms that the you have been approved by Service Canada to receive the Maternity/Parental benefits. 
>    - [_only if taking extended parental leave allowance_] Since you opted for extended parental leave allowance, you must provide another proof of approval for your parental leave allowance. You should complete the above steps again after you start your parental leave to claim parental leave allowance after you return.

**Comment by fazil-ey** [comment]
**Case 1: The applicant *did not* apply for allowance but is eligible**

- ****Are you applying for maternity leave allowance?”****
    - I do not want to claim any top-up allowance. I understand that this decision is irrevocable
- **AND**
- ****Are you applying for parental leave top-up allowance?****
    - I do not want to claim any top-up allowance. I understand that this decision is irrevocable

**Case 2: The applicant applied for allowance**

- ****Are you applying for maternity leave allowance?”****
    - I want to receive my top-up allowance during my leave after acceptance from EI
- **AND**
- ****Are you applying for parental leave top-up allowance?****
    - I want to receive my top-up allowance during my leave. I understand I will receive allowance only after my EI application is accepted

**Case 3: The applicant deferred allowance**

- ****Are you applying for maternity leave allowance?”****
    - I want to defer my decision to claim the allowance. I can choose to claim my allowance after I return to work. I understand that I need to return to work for at least the same amount of time I was on leave to claim the allowance. I also understand I will receive allowance only after my EI application is accepted
- **AND**
- ****Are you applying for parental leave top-up allowance?****
    - I want to defer my decision to claim top-up allowance. I can choose to claim my allowance after I return to work. I understand that I need to work for at least the same amount of time I was on leave to claim the allowance. I also understand I will receive allowance only after my EI application is accepted

**Case 4: The applicant is not eligible for benefits**

- Employee Information - Appointment Status K & L → not eligible (refer #612)

**Comment by bhumin-fw** [comment]
Progress:
Worked on "form submission" email and workflow.
Working on "On approval" emails and workflow. Email will be send based on application status and allowance + content of email will be design based on employee class.

**Comment by bhumin-fw** [comment]
@fazil-ey 
2.1.3 Case 3: The applicant deferred allowance

In above case? there is no option to select "extended allowance". But in template there is one bullet point says show only if extended allowance selected.

**Comment by fazil-ey** [comment]
Good catch! We don't need the extended allowance selection for deferred allowance case. Updated the ticket and removed the message

**Comment by bhumin-fw** [comment]
@fazil-ey 
Point 1 & 2 are ready to test on TEST and DEV environment.
Point 3: Required additional action like "not validated" , No additional button added yet in form. Workflow has been setup.
Point 4: Added a separate ticket to handle conditions and workflow timer [#1218](https://app.zenhub.com/workspaces/digital-journeys-61c10fe84c8bcb001491c5f5/issues/gh/bcgov/digital-journeys/1218)

**Comment by fazil-ey** [comment]
Hey @bhumin-fw ... some issues found. Happy to block some time and test this together next week if that helps 
 1.2 - the supervisor email should have the actual start date "The requested start date for the leave is **[start date]**" instead of the placeholder 
2.1.1 - applicant or supervisor are not receiving any emails after approval (i.e hitting validate submission)

**Comment by bhumin-fw** [comment]
@fazil-ey for sure, let's connect sometime next week and test it together.

**Comment by bhumin-fw** [comment]
@fazil-ey fix start date content in email.
I did receive mail after approval for all cases. It took 5-10minutes to send email with PDF. Let me know when is a good time to review that together, or let me know if you still have same issue about not receiving an email.

**Comment by fazil-ey** [comment]
Testing results tested 1 & 2 worked on Jul 21. However i am noticing some inconsistencies with the emails for some workflows e.g. different choices. Will retest this again when testing 3 & 4 points in separate tickets. Closing this for now
